src/js/components/purchaseOrder/PurchaseOrderListFilters.jsx

```diff
@@ -134,14 +133,9 @@ const PurchaseOrderListFilters = ({
   // Purchasing organizations (organizations with ROLE_BUYER)
   const [defaultValues, setDefaultValues] = useState({});
   const isCentralPurchasingEnabled = supportedActivities.includes('ENABLE_CENTRAL_PURCHASING');
-  const prevLocation = usePrevious(currentLocation);
-  const [initiallyFetched, setInitiallyFetched] = useState(false);
 
   const determineDefaultValues = () => {
-    // Avoid unnecessary re-fetches if getAppContext triggers fetching session info
-    // but currentLocation doesn't change
-    // eslint-disable-next-line max-len
-    if ((!initiallyFetched && currentLocation?.id) || (prevLocation && prevLocation.id !== currentLocation?.id)) {
+    if (currentLocation?.id && buyers) {
```

Comment:
had to include `buyers` there as well, because it could also trigger refetch (you could have currentLocation.id and fetch, but the buyers fetch was still in progress and it caused another refetch then).